File: core/model/repository.go
package model

type ChangeSet struct {
	Incoming     PriceUpdated
	UpdatedStock Stock
	Outgoing     StockUpdated
	OldStockRev  string
}

type Repository interface {
	PriceUpdatedEventExists(eventID string) (bool, error)
	LoadStock(stockID string) (Stock, error)
	Submit(changeSet ChangeSet) error
}

File: stocks/src/core/model/repository.ts
import { Stock } from "./stock";
import { PriceUpdated, StockUpdated } from "./events";

export interface ChangeSet {
  readonly incoming: PriceUpdated;
  readonly updatedStock: Stock;
  readonly outgoing: StockUpdated;
  readonly oldStockRev: string;
}

export interface Repository {
  priceUpdatedEventExists(eventId: string): Promise<boolean>;
  loadStock(stockId: string): Promise<Stock>;
  submit(changeSet: ChangeSet): Promise<void>;
}
